minijinja/src/filters.rs

```diff
@@ -537,13 +535,15 @@ mod builtins {
     /// ```
     #[cfg_attr(docsrs, doc(cfg(feature = "builtins")))]
     pub fn first(value: Value) -> Result<Value, Error> {
-        match value.0 {
-            ValueRepr::String(s, _) => Ok(s.chars().next().map_or(Value::UNDEFINED, Value::from)),
-            ValueRepr::Seq(ref s) => Ok(s.first().cloned().unwrap_or(Value::UNDEFINED)),
-            _ => Err(Error::new(
+        if let Some(s) = value.as_str() {
+            Ok(s.chars().next().map_or(Value::UNDEFINED, Value::from))
+        } else if let Some(s) = value.as_seq() {
+            Ok(s.get_item(0).unwrap_or(Value::UNDEFINED))
```

Comment:
In case parity with the previous line is desired:

```suggestion
            Ok(s.iter().next().unwrap_or(Value::UNDEFINED))
```